qcodes/tests/test_snapshot.py

```diff
@@ -0,0 +1,39 @@
+"""
+Test module for snapshots of instruments and parameters
+"""
+
+from qcodes.tests.instrument_mocks import SnapShotTestInstrument
+
+
+def test_snapshot_skip_params_update(request):
+    """
+    Test that params_to_skip_update works as expected, in particular that only
+    the parameters given by that variable are skipped
+    """
+
+    params = ['v1', 'v2', 'v3', 'v4']
+
+
+    inst = SnapShotTestInstrument('inst',
+                                  params=params,
+                                  params_to_skip=[])
+
+    request.addfinalizer(inst.close)
```

Comment:
i'd move these two lines inside the loop instead of
```
inst._params_to_skip = param_to_skip
inst.reset_counter()
```
in order to reduce unnecessary state management.